node_modules/@npmcli/arborist/lib/arborist/audit.js

```diff
@@ -34,6 +34,20 @@ module.exports = cls => class Auditor extends cls {
       options.filterSet =
         this.excludeWorkspacesDependencySet(tree)
     }
+
+    if (options.package.length > 0) {
+      if (options.filterSet !== undefined) {
+        const nodes = tree.inventory.query('packageName', options.package)
+        options.filterSet.add(...nodes)
```

Comment:
should filter where `!options.package.includes(node.packageName)` against the filterSet, since it exists